src/ucp/wireup/wireup_cm.c

```diff
@@ -359,14 +454,82 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
 static ssize_t ucp_cm_server_priv_pack_cb(void *arg, const char *dev_name,
                                           void *priv_data)
 {
-    ucs_error("UCP CM wireup is not completely implemented");
-    return UCS_ERR_NOT_IMPLEMENTED;
+    ucp_wireup_sockaddr_data_t *sa_data = priv_data;
+    ucp_ep_h ep                         = arg;
+    ucp_worker_h worker                 = ep->worker;
+    ucp_ep_config_key_t key             = ucp_ep_config(ep)->key;
+    uint64_t tl_bitmap                  = 0;
+    ucp_wireup_ep_t *wireup_ep;
+    uct_cm_attr_t cm_attr;
+    void* ucp_addr;
+    size_t ucp_addr_size;
+    ucp_rsc_index_t lane;
+    ucs_status_t status;
+
+    UCS_ASYNC_BLOCK(&worker->async);
+
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        if (lane != ucp_ep_get_cm_lane(ep)) {
+            tl_bitmap |= UCS_BIT(ucp_ep_config(ep)->key.lanes[lane].rsc_index);
+        }
+    }
+
+    /* make sure that all lanes are created on correct device */
+    ucs_assert(ucp_context_dev_tl_bitmap(worker->context, dev_name) ==
+               (tl_bitmap | ucp_context_dev_tl_bitmap(worker->context,
+                                                      dev_name)));
```

Comment:
can be simplified

Reply:
how?

Reply:
`ucs_assert(!(tl_bitmap & ~ucp_context_dev_tl_bitmap(worker->context, dev_name)))`